src/renderer_gl.cpp

```diff
@@ -6083,6 +6083,26 @@ namespace bgfx { namespace gl
 				;
 		}
 
+		if (!s_renderGL->m_gles3
+		&& !s_extension[Extension::OES_texture_npot].m_supported
+		&& (!!(m_width & (m_width - 1)) || !!(m_height & (m_height - 1))))
+		{
+			// npot textures do not support trilinear filtering and must clamp to edge
+			_flags &= ~(0
+				| BGFX_SAMPLER_MIP_MASK
+				| BGFX_SAMPLER_U_MASK
+				| BGFX_SAMPLER_V_MASK
+				| BGFX_SAMPLER_W_MASK
+				);
+
+			_flags |= 0
+				| BGFX_SAMPLER_MIP_POINT
+				| BGFX_SAMPLER_U_CLAMP
+				| BGFX_SAMPLER_V_CLAMP
+				| BGFX_SAMPLER_W_CLAMP
+				;
```

Comment:
This part would ultimately produce wrong results unlike code above that just changes sampling from linear to point. Ideally in those cases it would be done via caps, and checked by `bgfx::isTextureValid`.

Reply:
You're right. I would argue that changing sampling in some cases also produces incorrect results (admittedly less dramatic than a change of wrap mode). 

However, as the default wrap mode is REPEAT, and because a very common use of texturing actually don't sample outside [0,1], it's very easy to forget to set CLAMP mode; this is actually how we stumbled on this. In this situation this patch would just make the texture work, without user side checks, and with limited impact to result (bilinear instead of trilinear).

But again, you may have a point design-wise.

Reply:
> Which platform / OS / HW?

WebGL 1. We have mixed web clients with some that have not yet switched to WebGL2. So this is not a huge deal for us, we can mostly wait for full support of WebGL2 which is not far away. I suppose someone could have the same issue on old Android phones and ES 2.

Reply:
I was thinking to force requirement to have that extension instead.

Reply:
> old Android phones and ES 2.

Yeah, people who have that old phones, are probably not good demographics to target for anything...

Reply:
Exactly. Anyways feel free to reject that PR, as we have ways around it in the meantime.